webrtc/RTCDataChannel-send-blob-order.html

```diff
@@ -0,0 +1,37 @@
+<!doctype html>
+<meta charset=utf-8>
+<title>RTCDataChannel.prototype.send for blobs</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="RTCPeerConnection-helper.js"></script>
+<script>
+promise_test(async t => {
+    [channel1, channel2] = await createDataChannelPair();
+
+    let done;
+    const promise = new Promise(resolve => done = resolve);
+
+    const data1 = new Blob(['blob']);
+    const data1Size = data1.size;
+    const data2 = new ArrayBuffer(8);
+    const data2Size = data2.byteLength;
+
+    let firstMessage = true;
+
+    channel2.binaryType = "arraybuffer";
+    channel2.onmessage = t.step_func((e) => {
+        if (firstMessage) {
+            assert_equals(e.data.byteLength, data1Size);
+            firstMessage = false;
+            return;
+        }
+        assert_equals(e.data.byteLength, data2Size);
+        done();
```

Comment:
This uses the deferred pattern, which we can avoid it with async/await. E.g.:
```js
    channel1.send(data1);
    channel1.send(data2);
    let e = await new Promise(r => channel2.onmessage = r);
    assert_equals(e.data.byteLength, data1Size);
    e = await new Promise(r => channel2.onmessage = r);
    assert_equals(e.data.byteLength, data2Size);
```
(I've [tested](https://jsfiddle.net/jib1/v2ctmg5L/) that it works) This also gets rid of needing the `t.step_func` function in a `promise_test`.

Reply:
@jan-ivar See https://github.com/w3c/webrtc-pc/issues/2215#issuecomment-526714892

Reply:
nice, updated accordingly